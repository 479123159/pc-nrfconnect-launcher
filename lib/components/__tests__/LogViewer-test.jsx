jest.mock('react-infinite', () => 'Infinite');

/* eslint-disable import/first */
import React from 'react';
import renderer from 'react-test-renderer';
import Immutable from 'immutable';
import LogViewer from '../LogViewer';
/* eslint-enable import/first */

function renderComponent(entries) {
    return renderer.create(<LogViewer logEntries={entries} autoScroll />).toJSON();
}

describe('LogViewer', () => {
    it('should render log entries', () => {
        const entries = Immutable.List([
            {
                id: 1,
                level: 2,
                time: '2017-02-03T12:41:36.020Z',
                message: 'Info message',
            }, {
                id: 2,
                level: 4,
                time: '2017-02-03T13:41:36.020Z',
                message: 'Error message',
            },
        ]);
        expect(renderComponent(entries)).toMatchSnapshot();
    });
});
