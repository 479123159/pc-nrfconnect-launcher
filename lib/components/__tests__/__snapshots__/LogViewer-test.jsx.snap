exports[`LogViewer should render log entries 1`] = `
<div
  className="log-wrap">
  <div
    className="log-header">
    <div
      className="log-header-text">
      Log
    </div>
    <div
      className="padded-row log-header-buttons" />
  </div>
  <Infinite
    autoScroll={true}
    className="infinite-log"
    containerHeight={155}
    elementHeight={20}
    infiniteLoadBeginEdgeOffset={135}>
    <div
      className="log-entry log-info">
      <div
        className="time">
        13:41:36.0200
      </div>
      <div
        className="message">
        Info message
      </div>
    </div>
    <div
      className="log-entry log-error">
      <div
        className="time">
        14:41:36.0200
      </div>
      <div
        className="message">
        Error message
      </div>
    </div>
  </Infinite>
</div>
`;
